File: JavaScript/8js 2/isDigit.js
/*
|--------------------------------------------------------------------------
| Is it a number?
|--------------------------------------------------------------------------
|
*/

function isDigit(s) {
   return s == parseFloat(s);
}


function isDigit(s) {
   return /^-?\d+(\.\d+)?$/.test(s);
}


function isDigit(s) {
   return !!str.trim() && !isNaN(str);
}


console.log(isDigit("s2324"), false);
console.log(isDigit("-234.4"), true);

File: Python/8py 2/isDigit.py
"""
|--------------------------------------------------------------------------
| Is it a number?
|--------------------------------------------------------------------------
|
"""

def isDigit(string):
   try:
      float(string)
      return True
   except:
      return False
   

def isDigit(string):
   return string.lstrip('-').replace('.','').isdigit()


from re import match
def isDigit(string):
   return bool(match(r"^[-+]?\d+\.?\d*?$", string))


print(isDigit("s2324"), False)
print(isDigit("-234.4"), True)